### PR TITLE
fix: resolve dependencies issues and adds proofs generation

### PR DESCRIPTION
**Pull request by jotabulacios**
This PR resolves a dependency error that previously required using an outdated version, leading to false results during proof verification.

Additionally, it integrates proof generation directly within the gnark_bls12_381_circuit.go and gnark_bn254_circuit.go files in the test_examples/bls12_381 and test_examples/bn254 folders, respectively

to create the proofs run 
```bash
make generate-plonk_bls12_381-proof
```

or 

```bash
make generate-plonk_bn254-proof
```

### CODE REVIEW

**Review by entropidelic — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -121,34 +121,45 @@ send-plonk_bls12_381-proof-loop: ## Send a PLONK BLS12_381 proof using the task
...
-		--public-input task_sender/test_examples/bls12_381/plonk_pub_input.pub \
-		--verification-key task_sender/test_examples/bls12_381/plonk.vk \
+		--proof task_sender/test_examples/gnark_plonk_bls12_381_script/plonk.proof \
+		--public-input task_sender/test_examples/gnark_plonk_bls12_381_script/plonk_pub_input.pub \
+		--verification-key task_sender/test_examples/gnark_plonk_bls12_381_script/plonk.vk \
 		--config config-files/config.yaml \
 		--interval 10 \
 		2>&1 | zap-pretty
 
+run-plonk_bls12_381-script: ## Run the gnark_plonk_bls12_381_script
```
Comment (entropidelic) [inline Makefile]: I think it is clearer about what it does if named something like
```
generate-plonk_bls12_381-proof
```

Same with the bn254 one

File: task_sender/test_examples/gnark_plonk_bn254_script/main.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	// use unsafekzg.NewSRS to generate the SRS and the Lagrange interpolation of the SRS
+	// Setup prepares the public data associated to a circuit + public inputs.
+	// The kzg SRS must be provided in canonical and lagrange form.
+	// For test purposes, see test/unsafekzg package. With an existing SRS generated through MPC in canonical form,
+
+	r1cs := ccs.(*cs.SparseR1CS)
+	srs, srsLagrangeInterpolation, err := unsafekzg.NewSRS(r1cs)
+	// srs, err := test.NewKZGSRS(r1cs)
```
Comment (taturosati) [inline task_sender/test_examples/gnark_plonk_bn254_script/main.go:52]: remove comment
```suggestion
```

File: task_sender/test_examples/gnark_plonk_bls12_381_script/main.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	// use unsafekzg.NewSRS to generate the SRS and the Lagrange interpolation of the SRS
+	// Setup prepares the public data associated to a circuit + public inputs.
+	// The kzg SRS must be provided in canonical and lagrange form.
+	// For test purposes, see test/unsafekzg package. With an existing SRS generated through MPC in canonical form,
+
+	r1cs := ccs.(*cs.SparseR1CS)
+	srs, srsLagrangeInterpolation, err := unsafekzg.NewSRS(r1cs)
+	// srs, err := test.NewKZGSRS(r1cs)
```
Comment (taturosati) [inline task_sender/test_examples/gnark_plonk_bls12_381_script/main.go:52]: Remove comment
```suggestion
```

File: go.mod [hunk 4]
```diff
@@ -4,15 +4,18 @@ go 1.22.2
...
 	github.com/urfave/cli/v2 v2.27.1
 	golang.org/x/crypto v0.18.0
 )
 
 require github.com/mattn/go-isatty v0.0.19 // indirect
 
+require (
+	github.com/consensys/gnark v0.10.0
+	github.com/consensys/gnark-crypto v0.12.2-0.20240215234832-d72fcb379d3e
+)
```
Comment (taturosati) [inline go.mod]: should go back up with all direct requirements

File: Makefile [hunk 5]
```diff
@@ -121,34 +121,45 @@ send-plonk_bls12_381-proof-loop: ## Send a PLONK BLS12_381 proof using the task
...
 		--config config-files/config.yaml \
 		--interval 10 \
 		2>&1 | zap-pretty
 
+generate-plonk_bls12_381-proof: ## Run the gnark_plonk_bls12_381_script
+	@echo "Running gnark_plonk_bls12_381 script..."
+	@go run task_sender/test_examples/gnark_plonk_bls12_381_script/main.go
+
+
+
```
Comment (taturosati) [inline Makefile]: Remove excess space

```suggestion
```

**Comment by taturosati** [comment]
Proof generation & verification work. Small nitpicks but lgtm

**Review by taturosati — APPROVED** [review]
lgtm

**Pull request merged**